### PR TITLE
Port merge tests to compatibility suite

### PR DESCRIPTION
**Pull request by Mats-SX**

### CODE REVIEW

**Comment by davidegrohmann** [comment]
@Mats-SX how do we verify that the db has been updated correctly?

**Comment by Mats-SX** [comment]
The tests don't cover that. I find that their names convey what they are testing.

File: community/cypher/compatibility-suite/src/test/resources/cypher/MergeAcceptance.feature [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+Feature: MergeAcceptanceTest
+
+  Scenario: identifiers of deleted nodes should not be able to cause errors in later merge actions that do not refer to them
+    Given init: CREATE (:A)-[:R]->(:B)-[:R]->(:C)
+    When running: MATCH (a:A) -[ab]-> (b:B) -[bc]-> (c:C) DELETE ab, bc, b, c MERGE (newB:B) MERGE (a) -[:REL]->  (newB) MERGE (newC:C) MERGE (newB) -[:REL]-> (newC);
```
Comment (boggle) [inline community/cypher/compatibility-suite/src/test/resources/cypher/MergeAcceptance.feature]: Could we agree on a common style of how we write queries in the acceptance suite? I think capital keywords, and no spaces around arrows is the blessed way that we use in the manual.

**Comment by davidegrohmann** [comment]
@Mats-SX I don't think the new tests are testing the same thing as the old ones.

**Comment by Mats-SX** [comment]
It's a bit subtle. The first test just tests that it can execute without exceptions. The second and third tests, since they can not assert something like "this does not exist in the results", rely on the fact that the created nodes does not have properties, while the original nodes (which are deleted) do. If one of the original nodes would have turned up in the results, properties would have been written out, causing the assertion to fail.

I don't mind closing this PR now, if you think the compatibility suite is unsuited (heh) for these test cases.

**Comment by davidegrohmann** [comment]
@Mats-SX let's close this until we have a good understanding on how to tests updates in the compatibility suite.

**Comment by craigtaverner** [comment]
I wonder if we could have a 'then' syntax which asserts that a specific query has results. Like we had the init to create something before the test, we have a post-test query to check the contents of the database?

**Pull request closed without merging**